spec/integration/ffmpeg_adapter_spec.rb

```diff
@@ -428,4 +428,103 @@ def find_encode(id)
       expect { running_job.cancel! }.to raise_error(ActiveEncode::CancelError)
     end
   end
+
+  describe "#remove_old_files!" do
+    subject { created_job }
+    # 'exit_status.code' and 'progress' seem to be hidden files so rspec does not see them.
+    # That is why they are not explicitly included in the tests even though they are in the filenames list.
+    # If they were not being deleted they would cause other tests to fail.
+    let(:filenames) { ['input_metadata', 'error.log', 'pid', 'exit_status.code', 'progress'] }
+    let(:pathnames) { filenames.each_with_index { |fn, i| filenames[i] = fn.dup.prepend("#{work_dir}/#{subject.id}/") } }
+
+    # There was some flaky behavior with the file creation for created_job that
+    # would cause tests to fail. This ensures the files are created.
+    before :each do
+      FileUtils.touch(pathnames)
+    end
+
+    context ":no_outputs" do
+      it "deletes files created from encode process older than 2 weeks" do
+        # Another measure to give files time to be created.
+        sleep 1
+        travel 3.weeks do
+          expect { described_class.remove_old_files! }
+            .to change { File.exist?(pathnames[0]) }.from(true).to(false)
+            .and change { File.exist?(pathnames[1]) }.from(true).to(false)
+            .and change { File.exist?(pathnames[2]) }.from(true).to(false)
```

Comment:
I think this would be easier to read if each pathname was given its own `let` declaration and `pathnames` became an array of these variables.
Something like:
```
let(:base_path) { "#{work_dir}/#{subject.id}" }
let(:input_metadata_file) { "#{base_path}/input_metadata" }
...
let(:pathnames) { [input_metadata_file, error_log_file, pid_file, exit_status_file, progress_file] }
```
Then these lines could become:
```suggestion
            .to change { File.exist?(input_metadata_file) }.from(true).to(false)
            .and change { File.exist?(error_log_file) }.from(true).to(false)
            .and change { File.exist?(pid_file) }.from(true).to(false)
```